pkg/apis/core/validation/validation.go

```diff
@@ -3720,10 +3756,69 @@ func ValidatePodCreate(pod *core.Pod, opts PodValidationOptions) field.ErrorList
 	if len(pod.Spec.EphemeralContainers) > 0 {
 		allErrs = append(allErrs, field.Forbidden(fldPath.Child("ephemeralContainers"), "cannot be set on create"))
 	}
+	allErrs = append(allErrs, ValidateSeccompAnnotationsAndFields(pod.ObjectMeta, &pod.Spec, field.NewPath(""))...)
```

Comment:
this should also be called from ValidatePodTemplateSpec so we validate annotation/podSpec combinations in workload objects (add a unit test for that)